jgiven-core/src/main/java/com/tngtech/jgiven/report/ReportGenerator.java

```diff
@@ -43,39 +44,39 @@ public String formatName() {
      * Starts the respective report (default is HTML5)
      */
     public void generate( String... args ) {
-        Format format = ConfigOptionParser.getFormat( args );
+        var format = ConfigOptionParser.getFormat( args );
         switch( format ) {
             case ASCIIDOC:
-                new AsciiDocReportGenerator().generateFromCommandLine( args );
+                loadReportGenerator(ASCIIDOC_GENERATOR_FQCN).generateFromCommandLine(args);
                 break;
             case TEXT:
                 new PlainTextReportGenerator().generateFromCommandLine( args );
                 break;
-            case HTML:
-            case HTML5:
+            case HTML, HTML5:
             default:
-                ReportGenerator.generateHtml5Report().generateFromCommandLine( args );
+                ReportGenerator.loadReportGenerator(HTML5_REPORT_GENERATOR_FQCN).generateFromCommandLine(args);
                 break;
         }
     }
 
     /**
-     * Searches the Html5ReportGenerator in Java path and instantiates the report
-     */
-    public static AbstractReportGenerator generateHtml5Report() {
-        AbstractReportGenerator report;
-        try {
-            Class<?> aClass = ReportGenerator.class.getClassLoader()
-                    .loadClass( "com.tngtech.jgiven.report.html5.Html5ReportGenerator" );
-            report = (AbstractReportGenerator) aClass.getDeclaredConstructor().newInstance();
+	 * Searches the Html5ReportGenerator in Java path and instantiates it
+	 */
+	public static AbstractReportGenerator loadHtml5ReportGenerator() {
```

Comment:
is that one even used?

Reply:
It is a bit vestigial and I would rather remove it in upcoming changes, but there are indeed still three usages.